app/javascript/hooks/use-history.ts

```diff
@@ -11,28 +11,36 @@ function removeEmpty<TParams>(obj: TParams) {
         return v !== null
       }
     })
-    .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {})
+    .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {}) as TParams
 }
 
 function toQuery<TParams>(params: TParams): string {
```

Comment:
```suggestion
function toQuery<TParams extends Record<string, unknown>>(params: TParams): string {
```

I think this is what you meant?